File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/decorator/BigTrouser.java
package jv.com.tencent.cain.design_pattern.decorator;

/**
 * @author cainjiang
 * @date 2019-03-31
 * 具体的装饰者
 */
public class BigTrouser extends Finery {
    @Override
    protected void show() {
        System.out.print("垮裤 ");
        super.show();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/decorator/BigTrouser.kt
package kt.com.tencent.cain.desigin_pattern.decorator

/**
 * @author cainjiang
 * @date 2019-03-31
 * 具体的装饰者
 */
class BigTrouser : Finery() {
    override fun show() {
        print("垮裤 ")
        super.show()
    }
}